src/main/java/org/junitpioneer/jupiter/StopwatchExtension.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import static java.lang.System.currentTimeMillis;
+
+import java.time.Clock;
+
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
+import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+
+/**
+ * The StopwatchExtension implements callback methods for the {@code @Stopwatch} annotation.
+ */
+class StopwatchExtension implements BeforeTestExecutionCallback, AfterTestExecutionCallback {
+
+	private final Clock clock = Clock.systemUTC();
+	private static final Namespace NAMESPACE = Namespace.create(StopwatchExtension.class);
+
+	@Override
+	public void beforeTestExecution(ExtensionContext context) {
+		if (PioneerAnnotationUtils.isAnyAnnotationPresent(context, Stopwatch.class)) {
```

Comment:
I honestly think either this check is unnecessary or should throw an `ExtensionConfigurationException`.
This extension can be invoked by two things:
- Adding the `@Stopwatch` annotation (which has `@ExtendWith(StopwatchExtension.class)`)
- Adding `@ExtendWith(StopwatchExtension.class)` to your test class

If the explicit registration is not considered bad configuration then this check is unnecessary.
If the explicit registration is considered bad configuration then this check should be the other way around, i.e.:
```
if (!PioneerAnnotationUtils.isAnyAnnotationPresent(context, Stopwatch.class) {
    throw new ExtensionConfigurationException(/* Meaningful message */);
}
storeNowAsLaunchTime(context);
```

Reply:
An additional note - you could write a test for this. Have a class annotated with `@ExtendWith(StopwatchExtension.class)` and a `@Test` not annotated with `@Stopwatch`. Execute it with the testkit and the result should be a failed test and an `ExtensionConfigurationException`.